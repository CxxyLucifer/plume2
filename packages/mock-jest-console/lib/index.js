"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseMock = /** @class */ (function () {
    function BaseMock() {
        this._logs = [];
        this.mock();
    }
    BaseMock.prototype.mock = function () { };
    BaseMock.prototype.mockFn = function () {
        var _this = this;
        return jest.fn(function (log) { return _this._logs.push(log); });
    };
    Object.defineProperty(BaseMock.prototype, "logs", {
        get: function () {
            return this._logs;
        },
        enumerable: true,
        configurable: true
    });
    BaseMock.prototype.reset = function () {
        this._logs = [];
    };
    return BaseMock;
}());
exports.BaseMock = BaseMock;
var MockWarn = /** @class */ (function (_super) {
    __extends(MockWarn, _super);
    function MockWarn() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockWarn.prototype.mock = function () {
        console.warn = this.mockFn();
    };
    return MockWarn;
}(BaseMock));
exports.MockWarn = MockWarn;
var MockLog = /** @class */ (function (_super) {
    __extends(MockLog, _super);
    function MockLog() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockLog.prototype.mock = function () {
        console.log = this.mockFn();
    };
    return MockLog;
}(BaseMock));
exports.MockLog = MockLog;
var MockDir = /** @class */ (function (_super) {
    __extends(MockDir, _super);
    function MockDir() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockDir.prototype.mock = function () {
        console.dir = this.mockFn();
    };
    return MockDir;
}(BaseMock));
exports.MockDir = MockDir;
var MockDebug = /** @class */ (function (_super) {
    __extends(MockDebug, _super);
    function MockDebug() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockDebug.prototype.mock = function () {
        console.debug = this.mockFn();
    };
    return MockDebug;
}(BaseMock));
exports.MockDebug = MockDebug;
var MockError = /** @class */ (function (_super) {
    __extends(MockError, _super);
    function MockError() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockError.prototype.mock = function () {
        console.error = this.mockFn();
    };
    return MockError;
}(BaseMock));
exports.MockError = MockError;
var MockTrace = /** @class */ (function (_super) {
    __extends(MockTrace, _super);
    function MockTrace() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockTrace.prototype.mock = function () {
        console.trace = this.mockFn();
    };
    return MockTrace;
}(BaseMock));
exports.MockTrace = MockTrace;
var MockConsole = /** @class */ (function (_super) {
    __extends(MockConsole, _super);
    function MockConsole() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    MockConsole.prototype.mock = function () {
        console.warn = this.mockFn();
        console.log = this.mockFn();
        console.dir = this.mockFn();
        console.debug = this.mockFn();
        console.error = this.mockFn();
        console.trace = this.mockFn();
    };
    return MockConsole;
}(BaseMock));
exports.MockConsole = MockConsole;
