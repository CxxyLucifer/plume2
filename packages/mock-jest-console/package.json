{
  "name": "mock-jest-console",
  "version": "1.1.1",
  "description": "jest.fn mock console.xxx",
  "main": "./lib/index.js",
  "types": "./lib/typings/index.d.ts",
  "keywords": [
    "mock jest console"
  ],
  "author": "hufeng",
  "license": "MIT"
}
