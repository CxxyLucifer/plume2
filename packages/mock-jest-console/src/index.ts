declare const jest: { fn: Function };

export class BaseMock {
  constructor() {
    this._logs = [];
    this.mock();
  }

  private _logs;

  mock() {}

  mockFn() {
    return jest.fn(log => this._logs.push(log));
  }

  get logs() {
    return this._logs;
  }

  reset() {
    this._logs = [];
  }
}

export class MockWarn extends BaseMock {
  mock() {
    console.warn = this.mockFn();
  }
}

export class MockLog extends BaseMock {
  mock() {
    console.log = this.mockFn();
  }
}

export class MockDir extends BaseMock {
  mock() {
    console.dir = this.mockFn();
  }
}

export class MockDebug extends BaseMock {
  mock() {
    console.debug = this.mockFn();
  }
}

export class MockError extends BaseMock {
  mock() {
    console.error = this.mockFn();
  }
}

export class MockTrace extends BaseMock {
  mock() {
    console.trace = this.mockFn();
  }
}

export class MockGroup extends BaseMock {
  mock() {
    console.group = this.mockFn();
  }
}

export class MockGroupCollapsed extends BaseMock {
  mock() {
    console.groupCollapsed = this.mockFn();
  }
}

export class MockTime extends BaseMock {
  mock() {
    console.time = this.mockFn();
  }
}

export class MockConsole extends BaseMock {
  mock() {
    console.warn = this.mockFn();
    console.log = this.mockFn();
    console.dir = this.mockFn();
    console.debug = this.mockFn();
    console.error = this.mockFn();
    console.trace = this.mockFn();
    console.group = this.mockFn();
    console.groupCollapsed = this.mockFn();
    console.time = this.mockFn();
  }
}
