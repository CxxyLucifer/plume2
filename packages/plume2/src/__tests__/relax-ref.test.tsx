import { MockLog } from 'mock-jest-console';
import React from 'react';
import renderer from 'react-test-renderer';
import { Actor, Relax, Store, StoreProvider } from '../';

class HelloActor extends Actor {
  defaultState() {
    return {
      hello: 'hello'
    };
  }
}

class AppStore extends Store {
  bindActor() {
    return [HelloActor];
  }
}

@StoreProvider(AppStore, { debug: true })
class TestApp extends React.Component {
  _text: any;

  componentDidMount() {
    expect(this._text._relaxProxy.props.relaxProps).toEqual({
      hello: 'hello'
    });
  }

  render() {
    return <Text ref={text => (this._text = text)} />;
  }
}

@Relax
class Text extends React.Component {
  props: {
    relaxProps?: {
      hello: string;
    };
  };

  static relaxProps = ['hello'];

  render() {
    return <div>{this.props.relaxProps.hello}</div>;
  }
}

it('test relax proxy ref', () => {
  const mockLog = new MockLog();
  const tree = renderer.create(<TestApp />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(mockLog.logs).toMatchSnapshot();
});
