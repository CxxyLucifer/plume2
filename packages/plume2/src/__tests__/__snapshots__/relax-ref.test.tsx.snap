// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test relax proxy ref 1`] = `
<div>
  hello
</div>
`;

exports[`test relax proxy ref 2`] = `
Array [
  "%cplume2@1.1.1ðŸš€",
  "===========store init==========",
  "Actor:reduce HelloActor",
  "StoreProvider(TestApp) will mount ðŸš€",
  "Relax(Text) willMount rx store true ðŸš€",
  "props => {
  \\"relaxProps\\": {
    \\"hello\\": \\"hello\\"
  }
}",
]
`;
